### PR TITLE
Allow amp-list with only [src] attribute

### LINKED ISSUE #12372: amp-list: Remove src attribute requirement?
Context: #12367

For some use cases of amp-list (i.e. with amp-bind), having `src` attribute is not strictly necessary. To workaround, pubs can just set a garbage value but amp-list still attempts an XHR.

Should we (1) remove that requirement outright or (2) only allow omission when `[src]` is present or (3) require existence of placeholder element?

/cc @4-1-1 @ericlindley-g

### PR DESCRIPTION
**Pull request by choumx**
Fixes #12372.

- Dupes the tag spec with `[src]` being mandatory and `src` being optional instead.

FR for a less ugly implementation: #14006

### CODE REVIEW

File: extensions/amp-list/0.1/test/validator-amp-list.html [hunk 1]
```diff
@@ -29,12 +29,17 @@
...
-  <!-- Example of a valid amp-list -->
+  <!-- Valid: amp-list with only src attribute -->
   <amp-list width=10 height=10
             src="https://data.com/articles.json?ref=CANONICAL_URL">
     <div></div>
   </amp-list>
-  <!-- Valid amp-list with amp-bind attributes -->
+  <!-- Valid: amp-list with only [src] attribute -->
+  <amp-list width=10 height=10
+            src="https://data.com/articles.json?ref=CANONICAL_URL">
```
Comment (honeybadgerdontcare) [inline extensions/amp-list/0.1/test/validator-amp-list.html]: Replace `src` with `[src]`

File: extensions/amp-list/0.1/test/validator-amp-list.out [hunk 2]
```diff
@@ -30,12 +30,17 @@ FAIL
...
-|    <!-- Example of a valid amp-list -->
+|    <!-- Valid: amp-list with only src attribute -->
 |    <amp-list width=10 height=10
 |              src="https://data.com/articles.json?ref=CANONICAL_URL">
 |      <div></div>
 |    </amp-list>
-|    <!-- Valid amp-list with amp-bind attributes -->
+|    <!-- Valid: amp-list with only [src] attribute -->
+|    <amp-list width=10 height=10
+|              src="https://data.com/articles.json?ref=CANONICAL_URL">
```
Comment (honeybadgerdontcare) [inline extensions/amp-list/0.1/test/validator-amp-list.out]: Same: `src` -> `[src]`

File: extensions/amp-list/validator-amp-list.protoascii [hunk 3]
```diff
@@ -60,7 +60,43 @@ tags: {  # <amp-list>
...
+    supported_layouts: FLEX_ITEM
+    supported_layouts: NODISPLAY
+    supported_layouts: RESPONSIVE
+  }
+}
+tags: {  # <amp-list> with mandatory [src] attr
+  html_format: AMP
+  html_format: EXPERIMENTAL
+  tag_name: "AMP-LIST"
+  spec_name: "AMP-LIST-[SRC]"
```
Comment (honeybadgerdontcare) [inline extensions/amp-list/validator-amp-list.protoascii]: Perhaps a space instead of a dash between `AMP-LIST` and `[SRC]`.

**Comment by choumx** [comment]
Thanks for the reviews guys!

**Comment by yajiv185** [comment]
it is still giving error (The mandatory attribute 'src' is missing in tag 'amp-list'. (see https://www.ampproject.org/docs/reference/components/amp-list)), Why is that?

**Comment by choumx** [comment]
@yajiv185 It takes a couple weeks for validator changes to propagate.

**Comment by Gregable** [comment]
These validator changes will be live everywhere within an hour.

**Comment by yajiv185** [comment]
@Gregable I think changes related to this feature is not live, right?(because still I getting error same error as above.)

**Pull request merged**